turbinia/lib/utils.py

```diff
@@ -0,0 +1,100 @@
+# -*- coding: utf-8 -*-
+# Copyright 2018 Google Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Common utils."""
+
+from __future__ import unicode_literals
+
+import os
+import subprocess
+import tempfile
+import threading
+
+
+def extract_artifacts(artifact_names, disk_path, output_dir):
+  """Extract artifacts using image_export from Plaso.
+
+  Args:
+    artifact_names: List of artifact definition names.
+    disk_path: Path to either a raw disk image or a block device.
+    output_dir: Path to directory to store the the extracted files.
+
+  Returns:
+    dict: file names and paths to extracted files.
+  """
+  # Plaso image_export expects artifact names as a comma separated string.
+  artifacts = ','.join(artifact_names)
+
+  image_export_cmd = [
+    'image_export.py',
+    '--artifact_filters', artifacts,
+    '--write', output_dir,
+    disk_path
+  ]
+
+  try:
+    subprocess.check_call(image_export_cmd)
+  except subprocess.CalledProcessError:
+    raise RuntimeError('image_export.py failed.')
+
+  collected_file_paths = []
+  for dirpath, _, filenames in os.walk(output_dir):
+    for filename in filenames:
+      collected_file_paths.append(os.path.join(dirpath, filename))
+
+  return collected_file_paths
+
+
+def bruteforce_password_hashes(password_hashes, timeout=90):
+  """Bruteforce password hashes using John the Ripper.
+
+  Args:
+    password_hashes (list): List of password hashes.
```

Comment:
Tiny nit, but maybe change it like s/hashes./hashes as strings./  just to disambiguate from the many other formats that hashes sometimes come in?

Reply:
Done